test/RightsService.Business.UnitTests/Commands/Role/CreateRoleCommandTests.cs

```diff
@@ -1,89 +1,143 @@
-        private readonly int _rightId = 123;
-        private readonly Guid _userId = Guid.NewGuid();
 
-        [OneTimeSetUp]
-        public void OneTimeSetUp()
+        [SetUp]
+        public void SetUp()
         {
-            _mocker = new AutoMocker();
-            _command = _mocker.CreateInstance<CreateRoleCommand>();
+            _autoMocker = new AutoMocker();
+
+            _items = new Dictionary<object, object>();
+            _items.Add("UserId", _userId);
 
             _newRequest = new CreateRoleRequest
             {
-                Name = "Create Smth",
-                Description = "Create smth in somewhere",
-                Rights = new List<int> { _rightId }
-            };
-
-            var dbUser = new DbUser {
-                UserId = _userId,
-                Role = new DbRole { Id = _roleId }
-            };
-            var dbRoleRight = new DbRoleRight {
-                Id = Guid.NewGuid(),
-                Role = new DbRole { Id = _roleId },
-                Right = new DbRight { Id = _rightId }
+                Name = "Name",
+                Description = "Description",
+                Rights = new List<int> {1}
             };
 
             _dbRole = new DbRole
             {
-                Id = Guid.NewGuid(),
-                Name = "Create Smth",
-                Description = "Create smth in somewhere",
-                CreatedAt = DateTime.UtcNow,
-                CreatedBy = _authorId,
-                Rights = new List<DbRoleRight> { dbRoleRight },
-                Users = new List<DbUser> { dbUser }
+                Id = _roleId
             };
 
-            _response = new OperationResultResponse<Guid>
+            _goodResponse = new OperationResultResponse<Guid>
             {
                 Body = _dbRole.Id,
                 Status = OperationResultStatusType.FullSuccess,
                 Errors = new List<string>()
             };
+
+            _autoMocker
+                .Setup<IHttpContextAccessor, IDictionary<object, object>>(x => x.HttpContext.Items)
+                .Returns(_items);
+
+            _autoMocker
+                .Setup<IRoleRepository, Guid>(x => x.Create(_dbRole))
+                .Returns(_roleId);
+
+            _autoMocker
+                .Setup<ICreateRoleRequestValidator, bool>(x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(true);
+
+            _autoMocker
+                .Setup<IDbRoleMapper, DbRole>(x => x.Map(It.IsAny<CreateRoleRequest>(), It.IsAny<Guid>()))
+                .Returns(_dbRole);
+            _autoMocker
+                .Setup<IDbRoleMapper, DbRole>(x => x.Map(It.Is<CreateRoleRequest>(x => x == null), It.IsAny<Guid>()))
+                .Throws(new ArgumentNullException("CreateRoleRequest"));
+
+            _autoMocker
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
+                .Returns(true);
+
+            _command = new CreateRoleCommand(
```

Comment:
use _command = _autoMock.CreateInstance<>(); ?

Reply:
i think, that explicit creating of this instance is better, because _autoMock.CreateInstance<CreateRoleCommand>() can create it with the real objects, if there aren't any mocks

Reply:
I think that explicit declaration of this entity is clearer

Reply:
and how it help you?